smhi/metobs.py

```diff
@@ -36,76 +41,109 @@ def __init__(self, type: str = "json"):
         self.table_raw = None
         self.table = None
 
-    def fetch_parameters(self, version: str = "1.0"):
+    def fetch_parameters(self, version: Union[str, int] = "1.0"):
         """
-        Fetch SMHI API parameters from version. Only supports `version = 1.0`.
+        Fetch SMHI MetObs API parameters from version. Only supports `version = 1.0`.
 
         Args:
             version: selected API version
         """
+        if version == 1:
+            version = "1.0"
+
         if version != "1.0":
-            raise Exception("Only supports version = 1.0.")
+            raise NotImplementedError(
+                "Version {} not supported. Only supports version = 1.0.".format(version)
+            )
 
         self.version = version
-        self.parameter = SMHIParameterV1(self.type, self.available_versions, version)
+        self.parameter = MetObsParameterV1(self.available_versions)
 
     def fetch_stations(self, parameter: str = None, parameter_title: str = None):
         """
-        Fetch SMHI API (version 1) stations from given parameter.
+        Fetch SMHI MetObs API (version 1) stations from given parameter.
 
         Args:
             parameter: id of data
             parameter_title: exact title of data
         """
         if parameter is None and parameter_title is None:
-            raise Exception("Both arguments None.")
-
-        if self.parameter is None:
-            raise Exception("Fetch parameters first.")
-
-        if parameter_title is None:
-            self.station = SMHIStationV1(
-                self.type, self.parameter.resource, parameter=parameter
-            )
+            raise NotImplementedError("Both arguments None.")
 
-        if parameter is None:
-            self.station = SMHIStationV1(
-                self.type, self.parameter.resource, parameter_title=parameter_title
+        if parameter:
+            self.station = MetObsStationV1(self.parameter.resource, parameter=parameter)
+        else:
+            self.station = MetObsStationV1(
+                self.parameter.resource, parameter_title=parameter_title
             )
 
-    def fetch_periods(self, station: str = None, station_set: str = None):
+    def fetch_periods(self, station: str = None, stationset: str = None):
```

Comment:
def get_periods(self, station: str = None, stationset: str = None):

Reply:
Fixed in e2356c1.